src/utils/register-foreign-multiple.coffee

```diff
@@ -0,0 +1,65 @@
+program = require 'commander'
+msgflo_nodejs = require 'msgflo-nodejs'
+fs = require 'fs'
+path = require 'path'
+foreigner = require '../foreign-participant'
+common = require '../common'
+yaml = require 'js-yaml'
+Promise = require 'bluebird'
+
+onError = (err) ->
+  console.log err
+  process.exit 1
+
+onComplete = ->
+  process.exit 0
+
+collectMap = (val, map) ->
+  [role, definition] = val.split ':'
+  map[role] = definition
+  return map
+
+main = ->
+  program
+    .option('--broker <uri>', 'Broker address', String, '')
+    .option('--roles <role:definition>', 'Map of roles and definition files', collectMap, {})
+    .option('--interval <SECONDS>', 'How often to send discovery message', Number, 60)
+    .option('--forever <true>', 'Keep running forever', Boolean, false)
+    .usage('[options]')
+    .parse(process.argv)
+  program = common.normalizeOptions program
+
+  readFile = Promise.promisify fs.readFile
+  messaging = msgflo_nodejs.transport.getClient program.broker
+  register = Promise.promisify foreigner.register.bind foreigner
+  Promise.map Object.keys(program.roles), (role) ->
+    defPath = path.resolve process.cwd(), program.roles[role]
+    readFile defPath, 'utf-8'
+    .then (contents) ->
+      return Promise.reject "No definition found in #{defPath}" unless contents
+      definition = yaml.safeLoad contents
+      return Promise.reject new Error ".component is not defined" unless definition.component
+      definition.role = role
+      definition.id = definition.role unless definition.id
+      definition = foreigner.mapPorts definition
```

Comment:
Where is `#ROLE` and such being replaced for queues? I don't see this anywere.
Needed to be compatible with `msgflo-register-foreign`, and use with .yaml files declaring a component which may have multiple instances.